src/include/processor/operator/persistent/index_builder.h

```diff
@@ -25,14 +25,21 @@ namespace processor {
 
 const size_t SHOULD_FLUSH_QUEUE_SIZE = 32;
 
+constexpr size_t BUFFER_SIZE = 64;
+using OptionalExtraDataBuffer = std::optional<common::StaticVector<WarningSourceData, BUFFER_SIZE>>;
+
+using OptionalWarningSourceData = std::optional<WarningSourceData>;
+
 class IndexBuilderGlobalQueues {
 public:
     explicit IndexBuilderGlobalQueues(transaction::Transaction* transaction,
         storage::NodeTable* nodeTable);
 
     template<typename T>
-    void insert(size_t index, storage::IndexBuffer<T> elem,
+    void insert(size_t index, storage::IndexBuffer<T> elem, OptionalExtraDataBuffer extraData,
```

Comment:
I'm thinking of making all the parameter data rvalue references to show that `insert` should take ownership of the data, something like:
```suggestion
    void insert(size_t index, storage::IndexBuffer<T> &&elem, OptionalExtraDataBuffer &&extraData,
```
@ray6080 do you think this is a good idea or does that make things more confusing? Maybe I should just go with the normal pattern of using const references for both?

Reply:
I've been using the copy-then-move pattern quite a bit and I feel like that one probably isn't the correct pattern to use.